runtime/domain.c

```diff
@@ -1321,7 +1350,9 @@ CAMLprim value caml_domain_spawn(value callback, value term_sync)
   }
   /* When domain 0 first spawns a domain, the backup thread is not active, we
      ensure it is started here. */
-  install_backup_thread(domain_self);
+  domain_self->tid = pthread_self();
+  if (atomic_load_acquire(&domain_self->backup_thread_running) == 0)
```

Comment:
The backup thread for the main domain is spawned lazily: not until you spawn a second domain. This check makes sure `install_backup_thread` is only invoked once for the first domain, rather than on every domain spawn.